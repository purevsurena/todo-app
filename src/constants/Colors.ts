/**
 * Below are the colors that are used in the app. The colors are defined in the light and dark mode.
 * There are many other ways to style your app. For example, [Nativewind](https://www.nativewind.dev/), [Tamagui](https://tamagui.dev/), [unistyles](https://reactnativeunistyles.vercel.app), etc.
 */

const tintColorLight = "#0a7ea4";
const tintColorDark = "#fff";

export const Colors = {
  light: {
    text: "#11181C",
    background: "#fff",
    tint: tintColorLight,
    icon: "#687076",
    tabIconDefault: "#687076",
    tabIconSelected: tintColorLight,
  },
  dark: {
    text: "#ECEDEE",
    background: "#151718",
    tint: tintColorDark,
    icon: "#9BA1A6",
    tabIconDefault: "#9BA1A6",
    tabIconSelected: tintColorDark,    
  },
  generic: {
    dark: "#000",
    background: '#2c2c2e',
    cardBG: '#1c1c1e',
    greyText: '#9E9EA5',
    blue: '#54a0ff',
    orange: '#ff5722',
    turquoise: '#00adb5',
    yellow: '#FFFF00',
  }
};
