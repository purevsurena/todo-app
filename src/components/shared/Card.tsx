import React from "react";
import { View, StyleSheet, Dimensions } from "react-native";
import { MaterialCommunityIcons } from "@expo/vector-icons";
import { ThemedText } from "@/components/shared/ThemedText";
import { ThemedView } from "@/components/shared/ThemedView";
import Click from "@/components/layouts/Click";
import { Colors } from "@/constants/Colors";

type MaterialCommunityIconName = keyof typeof MaterialCommunityIcons.glyphMap;

interface CardProps {
  iconName: MaterialCommunityIconName;
  iconColor: string;
  title: string | number;
  subtitle: string;
  onPress: () => void;
}

const Card: React.FC<CardProps> = ({
  iconName,
  iconColor,
  title,
  subtitle,
  onPress,
}) => {
  return (
    <Click onPress={onPress}>
      <ThemedView style={styles.cardContainer}>
        <View style={styles.header}>
          <MaterialCommunityIcons name={iconName} size={30} color={iconColor} />
          <ThemedText type="title">{title}</ThemedText>
        </View>
        <ThemedText type="subtitle" darkColor={iconColor}>
          {subtitle}
        </ThemedText>
      </ThemedView>
    </Click>
  );
};

const styles = StyleSheet.create({
  cardContainer: {
    padding: 12,
    paddingVertical: 16,
    paddingTop: 24,
    borderRadius: 16,
    width: Dimensions.get("screen").width / 2 - 24,
    backgroundColor: Colors.generic.cardBG,
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    paddingBottom: 12,
  },
});

export default Card;
