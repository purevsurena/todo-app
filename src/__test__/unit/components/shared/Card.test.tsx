import React from "react";
import { render } from "@testing-library/react-native";
import Card from "@/components/shared/Card";
import { Colors } from "@/constants/Colors";

describe("Card Component", () => {
  it("should render the card with the correct title and subtitle", () => {
    const { getByText } = render(
      <Card
        iconName="home"
        iconColor={Colors.generic.blue}
        title="Card Title"
        subtitle="Card Subtitle"
        onPress={() => {}}
      />,
    );

    expect(getByText("Card Title")).toBeTruthy();
    expect(getByText("Card Subtitle")).toBeTruthy();
  });

  it("matches the snapshot", () => {
    const { toJSON } = render(
      <Card
        iconName="home"
        iconColor={Colors.generic.blue}
        title="Card Title"
        subtitle="Card Subtitle"
        onPress={() => {}}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
