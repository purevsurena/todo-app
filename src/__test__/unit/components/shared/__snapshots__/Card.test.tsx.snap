// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "backgroundColor": "#1c1c1e",
        "borderRadius": 16,
        "padding": 12,
        "paddingTop": 24,
        "paddingVertical": 16,
        "width": 351,
      },
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#54a0ff",
            "fontSize": 30,
          },
          undefined,
          {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ó°‹œ
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "lineHeight": 32,
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Card Title
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#11181C",
        },
        undefined,
        undefined,
        undefined,
        {
          "fontSize": 20,
          "fontWeight": "bold",
        },
        undefined,
        undefined,
      ]
    }
  >
    Card Subtitle
  </Text>
</View>
`;
